src/python/pants/jvm/resolve/jvm_tool.py

```diff
@@ -38,12 +38,12 @@
 class JvmToolBase(Subsystem):
     """Base class for subsystems that configure a set of artifact requirements for a JVM tool."""
 
-    # Default version of the tool. (Subclasses must set.)
-    default_version: ClassVar[str]
+    # Default version of the tool. (Subclasses may set.)
+    default_version: ClassVar[str | None] = None
 
     # Default artifacts for the tool in GROUP:NAME format. The `--version` value will be used for the
     # artifact version if it has not been specified for a particular requirement. (Subclasses must set.)
-    default_artifacts: ClassVar[Sequence[str]]
+    default_artifacts: ClassVar[tuple[str, ...]]
```

Comment:
This fixes type inference for the case of the empty list, but also ensures that the field is immutable.